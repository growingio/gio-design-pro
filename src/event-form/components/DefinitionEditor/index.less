@import '../../style/variables.less';

.checkbox-group {
  [class*='gio-checkbox-wrapper']:last-child {
    margin-bottom: 0;
  }

  .checkbox-editor {
    [class*='gio-checkbox-wrapper'] {
      display: flex;
      align-items: center;
      width: 110px;
    }

    .editor {
      margin-left: 8px;
    }

    &:not(:last-child) {
      margin-bottom: @middle-space;
    }
  }
}
