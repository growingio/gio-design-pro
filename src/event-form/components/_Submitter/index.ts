import Submitter from './Submitter';

export { SubmitterProps } from './Submitter';
export default Submitter;
