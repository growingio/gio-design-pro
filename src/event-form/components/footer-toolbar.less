@import '../../stylesheets/index.less';

@footer-bar-prefix-cls: ~'@{component-prefix}-event-form-footer-bar';

.@{footer-bar-prefix-cls} {
  // position: fixed;
  // right: 0;
  // bottom: 0;
  z-index: 99;
  display: flex;
  width: 100%;
  align-items: center;
  padding: 0 24px;
  line-height: 44px;
  background: #fff;
  background-color: #fff;
  box-shadow: 0 -1px 0 0 #dfe4ee;
  height: 60px;

  &-left {
    flex: 1;
  }

  &-right {
    & > * {
      margin-right: 8px;

      &:last-child {
        margin: 0;
      }
    }
  }
}
