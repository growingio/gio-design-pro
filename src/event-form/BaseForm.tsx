import React, { useState, useRef } from 'react';
import { Form } from '@gio-design/components';
import { FormInstance, FormProps } from '@gio-design/components/es/components/form';
import usePrefixCls from '@gio-design/components/es/utils/hooks/use-prefix-cls';
import { ButtonProps } from '@gio-design/components/es/components/button';
import type { SubmitterProps } from './components/Submitter';
import Submitter from './components/Submitter';
import { CommonFormProps } from './interfaces';

export type BaseFormProps = {
  contentRender?: (
    items: React.ReactNode[],
    submitter: React.ReactElement<SubmitterProps> | undefined,
    form: FormInstance<any>
  ) => React.ReactNode;
} & Omit<FormProps, 'onFinish'> &
  CommonFormProps;

const BaseForm: React.FC<BaseFormProps> = (props) => {
  const { contentRender, children, submitter, form: userForm, ...rest } = props;

  const [form] = Form.useForm();
  const formRef = useRef<FormInstance>(userForm || form);

  const items = React.Children.toArray(children);
  const submitterProps: SubmitterProps = typeof submitter === 'boolean' || !submitter ? {} : submitter;
  const [loading, setLoading] = useState<ButtonProps['loading']>(false);
  /** 渲染提交按钮与重置按钮 */
  const submitterNode =
    submitter === false ? undefined : (
      <Submitter
        key="submitter"
        {...submitterProps}
        form={userForm || form}
        submitButtonProps={{
          loading,
          ...submitterProps.submitButtonProps,
        }}
      />
    );

  const content = contentRender ? contentRender(items, submitterNode, formRef.current) : items;
  const prefixCls = usePrefixCls('event-form');
  return (
    <Form
      className={`${prefixCls}`}
      onKeyPress={(event) => {
        if (event.key === 'Enter') {
          formRef.current?.submit();
        }
      }}
      form={userForm || form}
      {...rest}
      onFinish={async (values) => {
        if (!rest.onFinish) {
          return;
        }
        setLoading(true);
        await rest.onFinish(values as Record<string, any>);
        setLoading(false);
      }}
    >
      {content}
    </Form>
  );
};

export default BaseForm;
