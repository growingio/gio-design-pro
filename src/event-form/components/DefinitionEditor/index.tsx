/* eslint-disable eqeqeq */
/* eslint-disable no-return-assign */
/* eslint-disable jsx-a11y/label-has-associated-control */
import React, { useState } from 'react';
import { Input, Radio, Form, Button } from '@gio-design/components';
import useProxyStore from '../../../hooks/useProxyStore';
import useUpdate from '../../../hooks/useUpdate';
import EditorModal from '../ModalForm';
import QsCheckbox from './CheckboxWithEditButton';
// import { LimitCondition } from '../../../typing';
import './index.less';
// 定义元素时的限定条件
export interface LimitCondition {
  content?: string;
  index?: string;
  href?: string;
  contentType?: 'match_phrase' | '=';
}
export interface Props {
  onChange?: (defineInfo: LimitCondition) => void;
  initValue: LimitCondition;
  isNative?: boolean;
}

export interface TempValue {
  [K: string]: any;
}

const initStore = (initValue: LimitCondition) => {
  const checks = {
    content: !!initValue.content,
    index: !!initValue.index,
    href: !!initValue.href,
  };

  return {
    checks,
    definition: initValue,
    init: { ...initValue },
  };
};

const CheckboxEditorGroup: React.FC<Props> = (props) => {
  const { onChange, initValue = {}, isNative } = props;
  const [showContentEditor, setShowContentEditor] = useState(false);
  const [, setShowHrefEditor] = useState(false);
  const [tempValue, setTempValue] = useProxyStore<TempValue>({ input: '', radio: '' });
  const [store, updateStore] = useProxyStore(() => initStore(initValue));

  useUpdate(() => {
    if (onChange) {
      const { definition, checks } = store;
      const def: LimitCondition = {};
      if (checks.content) {
        def.content = definition.content;
        def.contentType = definition.contentType;
      }
      if (checks.href) def.href = definition.href;
      if (checks.index) def.index = definition.index;
      onChange(def);
    }
  }, [store]);

  useUpdate(() => {
    updateStore(initStore(initValue));
  }, [initValue]);

  /**
   * 用来更新checkbox的值
   */
  const doCheckboxChange = (key: string) => {
    return (e: any) => {
      const { checked } = e.target;

      // 取消时恢复默认值
      const { definition } = store;
      if (!checked) {
        // @ts-ignore
        definition[key] = store.init[key];
        if (key == 'content') {
          definition.contentType = store.init.contentType;
        }
      }
      const checks = { ...store.checks, [key]: checked };
      updateStore({ checks, definition } as any);
    };
  };

  return (
    <>
      <EditorModal title="编辑元素内容" visible={showContentEditor}>
        <div className="editor-modal-body">
          <div className={`editor-modal-label-item ${tempValue.input ? '' : 'has-error'}`}>
            <label style={{ marginBottom: 8 }}>
              <span>{tempValue.radio != '=' ? '元素内容包含' : '元素内容为'}</span>
              <span className="required">*</span>
            </label>
            <Input
              value={tempValue.input}
              placeholder="请填写元素内容"
              // eslint-disable-next-line no-return-assign
              onChange={(e) => {
                return (tempValue.input = e.target.value);
              }}
            />
            <span style={{ visibility: tempValue.input ? 'hidden' : 'visible' }} className="error-tip">
              元素内容不能为空
            </span>
          </div>
          <div className="editor-modal-label-item">
            <label style={{ marginBottom: 8 }}>
              <span>按内容</span>
            </label>
            <Radio.Group value={tempValue.radio} onChange={(e) => (tempValue.radio = e.target.value)}>
              <Radio style={{ fontWeight: 'normal' }} value="=">
                精准匹配
              </Radio>
              <Radio style={{ fontWeight: 'normal' }} value="match_phrase">
                模糊匹配
              </Radio>
            </Radio.Group>
          </div>
        </div>
      </EditorModal>

      <EditorModal
        title="编辑跳转链接"
        trigger={<Button type="primary">新建表单</Button>}
        modalProps={
          {
            // onClose: () => console.log('run'),
          }
        }
        onFinish={async (values: any) => {
          console.log(values);
          return true;
        }}
      >
        <div className="editor-modal-body">
          <Form.Item name="href" label="跳转链接为" required>
            <Input.TextArea
              value={tempValue.input}
              placeholder="请填写跳转链接"
              rows={3}
              onChange={(e) => setTempValue({ input: e.target.value, radio: '' })}
              wrapStyle={{ width: '100%', height: 'unset' }}
            />
          </Form.Item>
        </div>
      </EditorModal>

      <div className="checkbox-group">
        {store.init.content && (
          <QsCheckbox
            checked={store.checks.content}
            onChange={doCheckboxChange('content')}
            onClickEditor={() => {
              setShowContentEditor(true);
              setTempValue({ input: store.definition?.content, radio: store.definition?.contentType });
            }}
          >
            元素内容
          </QsCheckbox>
        )}
        {!!store.init.index && (
          <QsCheckbox style={{ width: 200 }} checked={store.checks.index} onChange={doCheckboxChange('index')}>
            元素位置第{+store.init.index + +(isNative || 0)}位
          </QsCheckbox>
        )}
        {store.init.href && (
          <QsCheckbox
            checked={store.checks.href}
            onChange={doCheckboxChange('href')}
            onClickEditor={() => {
              setShowHrefEditor(true);
              tempValue.input = store.definition?.href;
            }}
          >
            跳转链接
          </QsCheckbox>
        )}
      </div>
    </>
  );
};

export default CheckboxEditorGroup;
