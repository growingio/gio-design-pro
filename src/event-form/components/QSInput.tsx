/* eslint-disable react/no-array-index-key */
import { Form, Input, Button } from '@gio-design/components';
import { PlusOutlined } from '@gio-design/icons';
import React, { useEffect, useState } from 'react';
import { Kv } from '../utils';

interface Props {
  value?: Kv[];
  onChange?: (value: Kv) => void;
}

const QSInput = (props: Props) => {
  const { value } = props;
  const [querySet, setQuerySet] = useState<Kv[]>(value);
  useEffect(() => {
    setQuerySet(value);
  }, [value]);

  const validateRules = {
    queryKey: [
      { required: true, message: '参数不能为空' },
      { pattern: /^((?!\*).)*$/, message: '参数不支持通配' },
      {
        validateTrigger: 'onBlur',
        validator: async (_: any, v: string) => {
          if (v) {
            const kv = querySet?.filter((q) => q?.key === v);
            if (kv && kv.length > 1) {
              return Promise.reject(new Error('参数不能重复'));
            }
          }
          return Promise.resolve(true);
        },
      },
    ],
    queryValue: [{ required: true, message: '参数值不能为空' }],
  };

  return (
    <Form.List name="query">
      {(fields, { add, remove }) => (
        <>
          {fields.map((field) => (
            <div
              className="query-filed-wrap"
              key={`query_${field.key}`}
              style={{ display: 'flex', flex: 1, alignItems: 'flex-start', marginBottom: '8px' }}
            >
              <Form.Item
                className="query__filed"
                labelWidth={0}
                {...field}
                key={`key${field.key}`}
                name={[field.name, 'key']}
                fieldKey={[field.key, `key`]}
                validateTrigger={['onChange', 'onBlur']}
                rules={validateRules.queryKey}
              >
                <Input placeholder="请输入参数" />
              </Form.Item>
              <div style={{ padding: '8px' }}>=</div>
              <Form.Item
                className="query__filed"
                labelWidth={0}
                {...field}
                key={`value${field.key}`}
                name={[field.name, 'value']}
                fieldKey={[field.key, `value`]}
                validateTrigger={['onChange', 'onBlur']}
                rules={validateRules.queryValue}
              >
                <Input placeholder="请输入参数值" />
              </Form.Item>
              <div>
                <PlusOutlined
                  style={{ transform: 'rotate(45deg)', padding: '8px', cursor: 'pointer' }}
                  onClick={() => remove(field.name)}
                />
              </div>
            </div>
          ))}
          <Form.Item labelWidth={0} className="query-input-add-filed">
            <Button type="text" onClick={() => add()}>
              添加查询条件
            </Button>
          </Form.Item>
        </>
      )}
    </Form.List>
  );
};

export default QSInput;
