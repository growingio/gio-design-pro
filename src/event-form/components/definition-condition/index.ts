import DefinitionEditor from './DefinitionEditor';
import './style/index.less';

export default DefinitionEditor;
