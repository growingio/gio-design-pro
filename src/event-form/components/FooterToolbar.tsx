import React, { ReactElement, ReactNode } from 'react';
import classNames from 'classnames';
import usePrefixCls from '@gio-design/components/es/utils/hooks/use-prefix-cls';
import './footer-toolbar.less';
// import { RouteContext, RouteContextType } from '../index';
import ReactDOM from 'react-dom';

export interface FooterToolbarProps {
  extra?: React.ReactNode;
  style?: React.CSSProperties;
  className?: string;
  renderContent?: (props: FooterToolbarProps & { leftWidth?: string }, dom: ReactElement) => ReactNode;
}
const FooterToolbar: React.FC<FooterToolbarProps> = (props) => {
  const { children, className, extra, style, renderContent } = props;
  const prefixCls = usePrefixCls('event-form');
  const baseClassName = `${prefixCls}-footer-bar`;
  const width = '100%';

  const dom = (
    <>
      <div className={`${baseClassName}-left`}>{extra}</div>
      <div className={`${baseClassName}-right`}>{children}</div>
    </>
  );

  return (
    <div className={classNames(className, `${baseClassName}`)} style={{ width, ...style }}>
      {renderContent
        ? renderContent(
            {
              ...props,
              leftWidth: width,
            },
            dom
          )
        : dom}
    </div>
  );
};
type PortalFooterToolbarProps = FooterToolbarProps & { container?: Element };
const PortalFooterToolbar: React.FC<PortalFooterToolbarProps> = (props) => {
  const { container, ...rest } = props;
  if (container) {
    return ReactDOM.createPortal(<FooterToolbar {...rest} />, container);
  }
  return <FooterToolbar {...rest} />;
};
export default PortalFooterToolbar;
