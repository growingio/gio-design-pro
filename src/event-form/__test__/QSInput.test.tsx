/* eslint-disable no-console */
import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import { Form } from '@gio-design/components';
import QSInput from '../components/QSInput';
import BaseForm from '../BaseForm';
// import { mount, shallow, render } from 'enzyme';

jest.useFakeTimers();

describe('QSInput', () => {
  it('render QSInput', async () => {
    const handleChange = jest.fn();
    // const defaultValue = [{ key: 'q1', value: '1' }];
    const defaultValue = {
      definition: {
        query: [
          { key: 'name', value: 'hello' },
          { key: 'a', value: '3' },
        ],
      },
    };

    const renderer = (
      <BaseForm initialValues={defaultValue} onValuesChange={(v, av) => handleChange(av)}>
        <Form.Item label="查询条件">
          <div className="query-input">
            <QSInput value={defaultValue.definition.query} />
          </div>
        </Form.Item>
      </BaseForm>
    );
    // const { container } = render(renderer);
    render(renderer);
    expect(screen.queryAllByPlaceholderText('请输入参数')).toHaveLength(2);
    expect(screen.queryAllByPlaceholderText('请输入参数值')).toHaveLength(2);

    expect(screen.queryByText('添加查询条件')).toBeTruthy();

    await act(async () => {
      fireEvent.click(screen.queryByText('添加查询条件'));
    });
    expect(screen.queryAllByPlaceholderText('请输入参数')).toHaveLength(3);
    expect(screen.queryAllByPlaceholderText('请输入参数值')).toHaveLength(3);
    expect(handleChange).toHaveBeenCalledTimes(1);
    await act(async () => {
      fireEvent.focus(screen.queryAllByPlaceholderText('请输入参数')[2]);
      fireEvent.change(screen.queryAllByPlaceholderText('请输入参数')[2], { target: { value: 'a' } });
      fireEvent.blur(screen.queryAllByPlaceholderText('请输入参数')[2]);
      jest.runOnlyPendingTimers();
      // fireEvent.change(screen.queryAllByPlaceholderText('请输入参数值')[2], { target: { value: '2' } });
    });
    // expect(screen.queryByText('参数不能重复')).toBeTruthy();
    // expect(container).toMatchSnapshot();
    await act(async () => {
      fireEvent.change(screen.queryAllByPlaceholderText('请输入参数')[2], { target: { value: 'b' } });
      fireEvent.change(screen.queryAllByPlaceholderText('请输入参数值')[2], { target: { value: '2' } });
    });
    await act(async () => {
      fireEvent.click(screen.queryAllByLabelText('plus-outlined')[0]);
    });
    // expect(handleChange).toHaveBeenCalledTimes(4);
    expect(handleChange).toHaveBeenLastCalledWith({
      definition: {
        query: [
          { key: 'a', value: '3' },
          { key: 'b', value: '2' },
        ],
      },
    });
  });
});
