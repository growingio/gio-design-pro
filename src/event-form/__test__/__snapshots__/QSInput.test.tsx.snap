// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QSInput render QSInput 1`] = `
<div>
  <form
    class="gio-form gio-event-form gio-form-middle gio-form-horizontal"
  >
    <div
      class="gio-field"
    >
      <div
        class="gio-field-label"
      >
        <label
          title="查询条件"
        >
          查询条件
          
        </label>
      </div>
      <div
        class="gio-field-control"
      >
        <div
          class="gio-field-control-input"
        >
          <div
            class="query-input"
          >
            <div
              class="query-filed-wrap"
              style="display: flex; flex: 1; align-items: flex-start; margin-bottom: 8px;"
            >
              <div
                class="gio-field query__filed"
              >
                <div
                  class="gio-field-label"
                  style="width: 0px;"
                />
                <div
                  class="gio-field-control"
                >
                  <div
                    class="gio-field-control-input"
                  >
                    <div
                      class="gio-input gio-input--middle"
                    >
                      <input
                        class="gio-input__content"
                        id="definition_query_0_key"
                        placeholder="请输入参数"
                        type="text"
                        value="name"
                      />
                    </div>
                  </div>
                  <div
                    class="gio-field-message"
                    data-has-feedback="false"
                  />
                </div>
              </div>
              <div
                style="padding: 8px;"
              >
                =
              </div>
              <div
                class="gio-field query__filed"
              >
                <div
                  class="gio-field-label"
                  style="width: 0px;"
                />
                <div
                  class="gio-field-control"
                >
                  <div
                    class="gio-field-control-input"
                  >
                    <div
                      class="gio-input gio-input--middle"
                    >
                      <input
                        class="gio-input__content"
                        id="definition_query_0_value"
                        placeholder="请输入参数值"
                        type="text"
                        value="hello"
                      />
                    </div>
                  </div>
                  <div
                    class="gio-field-message"
                    data-has-feedback="false"
                  />
                </div>
              </div>
              <div>
                <span
                  aria-label="plus-outlined"
                  class="gio-icon"
                  role="img"
                  style="transform: rotate(45deg); padding: 8px; cursor: pointer;"
                >
                  <svg
                    fill="currentColor"
                    height="16px"
                    viewBox="0 0 14 14"
                    width="16px"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <defs>
                      <style />
                    </defs>
                    <g>
                      <g>
                        <path
                          d="M6 6.51A.54.54 0 006.51 6V.49a.49.49 0 011 0V6a.54.54 0 00.49.51h5.48a.49.49 0 010 1H8a.54.54 0 00-.51.49v5.48a.49.49 0 01-1 0V8A.54.54 0 006 7.49H.49a.49.49 0 010-1z"
                          fill="currentColor"
                          fill-rule="evenodd"
                          id="plus-outlined_svg__plus"
                        />
                      </g>
                    </g>
                  </svg>
                </span>
              </div>
            </div>
            <div
              class="query-filed-wrap"
              style="display: flex; flex: 1; align-items: flex-start; margin-bottom: 8px;"
            >
              <div
                class="gio-field query__filed"
              >
                <div
                  class="gio-field-label"
                  style="width: 0px;"
                />
                <div
                  class="gio-field-control"
                >
                  <div
                    class="gio-field-control-input"
                  >
                    <div
                      class="gio-input gio-input--middle"
                    >
                      <input
                        class="gio-input__content"
                        id="definition_query_1_key"
                        placeholder="请输入参数"
                        type="text"
                        value="a"
                      />
                    </div>
                  </div>
                  <div
                    class="gio-field-message"
                    data-has-feedback="false"
                  />
                </div>
              </div>
              <div
                style="padding: 8px;"
              >
                =
              </div>
              <div
                class="gio-field query__filed"
              >
                <div
                  class="gio-field-label"
                  style="width: 0px;"
                />
                <div
                  class="gio-field-control"
                >
                  <div
                    class="gio-field-control-input"
                  >
                    <div
                      class="gio-input gio-input--middle"
                    >
                      <input
                        class="gio-input__content"
                        id="definition_query_1_value"
                        placeholder="请输入参数值"
                        type="text"
                        value="3"
                      />
                    </div>
                  </div>
                  <div
                    class="gio-field-message"
                    data-has-feedback="false"
                  />
                </div>
              </div>
              <div>
                <span
                  aria-label="plus-outlined"
                  class="gio-icon"
                  role="img"
                  style="transform: rotate(45deg); padding: 8px; cursor: pointer;"
                >
                  <svg
                    fill="currentColor"
                    height="16px"
                    viewBox="0 0 14 14"
                    width="16px"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <defs>
                      <style />
                    </defs>
                    <g>
                      <g>
                        <path
                          d="M6 6.51A.54.54 0 006.51 6V.49a.49.49 0 011 0V6a.54.54 0 00.49.51h5.48a.49.49 0 010 1H8a.54.54 0 00-.51.49v5.48a.49.49 0 01-1 0V8A.54.54 0 006 7.49H.49a.49.49 0 010-1z"
                          fill="currentColor"
                          fill-rule="evenodd"
                          id="plus-outlined_svg__plus"
                        />
                      </g>
                    </g>
                  </svg>
                </span>
              </div>
            </div>
            <div
              class="query-filed-wrap"
              style="display: flex; flex: 1; align-items: flex-start; margin-bottom: 8px;"
            >
              <div
                class="gio-field query__filed"
              >
                <div
                  class="gio-field-label"
                  style="width: 0px;"
                />
                <div
                  class="gio-field-control"
                >
                  <div
                    class="gio-field-control-input"
                  >
                    <div
                      class="gio-input gio-input--middle"
                    >
                      <input
                        class="gio-input__content"
                        id="definition_query_2_key"
                        placeholder="请输入参数"
                        type="text"
                        value="name"
                      />
                    </div>
                  </div>
                  <div
                    class="gio-field-message"
                    data-has-feedback="false"
                  />
                </div>
              </div>
              <div
                style="padding: 8px;"
              >
                =
              </div>
              <div
                class="gio-field query__filed"
              >
                <div
                  class="gio-field-label"
                  style="width: 0px;"
                />
                <div
                  class="gio-field-control"
                >
                  <div
                    class="gio-field-control-input"
                  >
                    <div
                      class="gio-input gio-input--middle"
                    >
                      <input
                        class="gio-input__content"
                        id="definition_query_2_value"
                        placeholder="请输入参数值"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="gio-field-message"
                    data-has-feedback="false"
                  />
                </div>
              </div>
              <div>
                <span
                  aria-label="plus-outlined"
                  class="gio-icon"
                  role="img"
                  style="transform: rotate(45deg); padding: 8px; cursor: pointer;"
                >
                  <svg
                    fill="currentColor"
                    height="16px"
                    viewBox="0 0 14 14"
                    width="16px"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <defs>
                      <style />
                    </defs>
                    <g>
                      <g>
                        <path
                          d="M6 6.51A.54.54 0 006.51 6V.49a.49.49 0 011 0V6a.54.54 0 00.49.51h5.48a.49.49 0 010 1H8a.54.54 0 00-.51.49v5.48a.49.49 0 01-1 0V8A.54.54 0 006 7.49H.49a.49.49 0 010-1z"
                          fill="currentColor"
                          fill-rule="evenodd"
                          id="plus-outlined_svg__plus"
                        />
                      </g>
                    </g>
                  </svg>
                </span>
              </div>
            </div>
            <div
              class="gio-field query-input-add-filed"
            >
              <div
                class="gio-field-label"
                style="width: 0px;"
              />
              <div
                class="gio-field-control"
              >
                <div
                  class="gio-field-control-input"
                >
                  <button
                    class="gio-btn gio-btn-link"
                    type="button"
                  >
                    <span>
                      添加查询条件
                    </span>
                  </button>
                </div>
                <div
                  class="gio-field-message"
                  data-has-feedback="false"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="gio-field-message"
          data-has-feedback="false"
        />
      </div>
    </div>
  </form>
</div>
`;
